### ISSUE TITLE
Snowflake GROUP BY CUBE

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2758]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Snowflake has a `GROUP BY CUBE` clause: https://docs.snowflake.com/en/sql-reference/constructs/group-by-cube.html

Currently the cube keyword is being parsed as a function name, where it should instead be a keyword followed by bracketed delimited groupCube.

This led to L010 not understanding that `CUBE` should be included in the keyword capitalisation policy (n.b. the rule is working fine, it's just the parsing needs fixing). 

Consider example query from docs:
```sql
select state, city, sum((s.retail_price - p.wholesale_price) * s.quantity) as profit 
 from products as p, sales as s
 where s.product_id = p.product_id
 group by cube (state, city)
 order by state, city nulls last
 ;
```
run fix with L010 uppercase config:
```console
sqlfluff fix test.sql --dialect snowflake --rules L010
```

### Expected Behaviour

```sql
SELECT state, city, sum((s.retail_price - p.wholesale_price) * s.quantity) AS profit 
 FROM products AS p, sales AS s
 WHERE s.product_id = p.product_id
 GROUP BY CUBE (state, city)
 ORDER BY state, city NULLS LAST
 ;
```

### Observed Behaviour

```sql
SELECT state, city, sum((s.retail_price - p.wholesale_price) * s.quantity) AS profit 
 FROM products AS p, sales AS s
 WHERE s.product_id = p.product_id
 GROUP BY cube (state, city)
 ORDER BY state, city NULLS LAST
 ;
```

### How to reproduce

details above

### Dialect

Snowflake

### Version

```
>>> sqlfluff version
0.10.1
>>> python3.7 --version
Python 3.7.12
```

### Configuration

```
[sqlfluff:rules:L010]
# Keywords
capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
